examples/call-runtime/lib.rs

```diff
@@ -0,0 +1,326 @@
+#[derive(scale::Encode)]
+enum RuntimeCall {
+    /// This index can be found by investigating runtime configuration. You can check the pallet
+    /// order inside `construct_runtime!` block and read the position of your pallet (0-based).
+    ///
+    ///
+    /// [See here for more.](https://substrate.stackexchange.com/questions/778/how-to-get-pallet-index-u8-of-a-pallet-in-runtime)
+    #[codec(index = 4)]
+    Balances(BalancesCall),
+}
+
+#[derive(scale::Encode)]
+enum BalancesCall {
+    /// This index can be found by investigating the pallet dispatchable API. In your pallet code,
+    /// look for `#[pallet::call]` section and check `#[pallet::call_index(x)]` attribute of the
+    /// call. If these attributes are missing, use source-code order (0-based).
+    #[codec(index = 0)]
+    Transfer {
+        dest: MultiAddress<AccountId, ()>,
+        #[codec(compact)]
+        value: u128,
+    },
+}
+
+#[ink::contract]
+mod runtime_call {
+    use crate::{
+        BalancesCall,
+        RuntimeCall,
+    };
+
+    use ink::env::Error as EnvError;
+
+    /// A trivial contract with a single message, that uses `call-runtime` API for performing
+    /// native token transfer.
+    #[ink(storage)]
+    #[derive(Default)]
+    pub struct RuntimeCaller;
+
+    #[derive(Debug, PartialEq, Eq, scale::Encode, scale::Decode)]
+    #[cfg_attr(feature = "std", derive(scale_info::TypeInfo))]
+    pub enum RuntimeError {
+        CallRuntimeReturnedError,
+    }
+
+    impl From<EnvError> for RuntimeError {
+        fn from(e: EnvError) -> Self {
+            match e {
+                EnvError::CallRuntimeReturnedError => {
+                    RuntimeError::CallRuntimeReturnedError
+                }
+                _ => panic!("Unexpected error from `pallet-contracts`."),
+            }
+        }
+    }
+
+    impl RuntimeCaller {
+        /// The constructor is `payable`, so that during instantiation it can be given some tokens
+        /// that will be further transferred with `transfer_through_runtime` message.
+        #[ink(constructor, payable)]
+        pub fn new() -> Self {
+            Default::default()
+        }
+
+        /// Tries to transfer `value` from the contract's balance to `receiver`.
+        ///
+        /// Fails if:
+        ///  - called in the off-chain environment
+        ///  - the chain doesn't allow `call-runtime` API (`UnsafeUnstableInterface` is turned off)
+        ///  - the chain forbids contracts to call `Balances::transfer` (`CallFilter` is too
+        ///    restrictive)
+        ///  - after the transfer, `receiver` doesn't have at least existential deposit
+        ///  - the contract doesn't have enough balance
+        #[ink(message)]
+        pub fn transfer_through_runtime(
+            &mut self,
+            receiver: AccountId,
+            value: Balance,
+        ) -> Result<(), RuntimeError> {
+            self.env()
+                .call_runtime(&RuntimeCall::Balances(BalancesCall::Transfer {
```

Comment:
The contract account can be reaped with this call if its balance fails below ed. This is bad. And not handling that and saying nothing on that regard in docs is even worse.

Reply:
note added

Reply:
This is still a foot-shooting contract, which I don't think we should demonstrate to ink! developers as an example. 

There is the [ink_env::minimal_balance()](https://paritytech.github.io/ink/ink_env/fn.minimum_balance.html) function available with which you could get current existential deposit value and add a safety check to the contract message

Reply:
note removed